### PR TITLE
Update mysql to 1.6.9-p0, adds install config

### PR DESCRIPTION
**Pull request by caseymcc**
As a side effect target name changed from MySQL::client to MySQL::mysqlclient. Will update the wiki after PR accepted.

### CODE REVIEW

**Comment by ruslo** [comment]
Testing:
* https://travis-ci.org/ingenue/hunter/builds/241231296?utm_source=github_status&utm_medium=notification
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1830

**Comment by ruslo** [comment]
All Travis CI tests failed

**Comment by caseymcc** [comment]
Yea, saw that but I dont understand them. The first error in clang is,

    undefined reference to `__gxx_personality_v0'

which seems to some issue with libstdc++. But you can see right above that the call to clang including it

    /usr/local/clang-3.5.0/bin/clang -O3 -DNDEBUG  -stdlib=libstdc++ -rdynamic CMakeFiles/MySQL-client-example.dir/main.c.o  -o MySQL-client-example 

then a message letting you know clang is ignoring it

    clang: warning: argument unused during compilation: '-stdlib=libstdc++'

I don't think any of my changes should have affected any of this.

**Comment by ruslo** [comment]
Don't know what is the reason, may be it's C vs C++ issue?

**Comment by caseymcc** [comment]
I am also baffled as all of the test I ran before putting in the PR succeeded.
https://travis-ci.org/caseymcc/hunter/builds/240946709

comparing the logs:

    /usr/local/clang-3.5.0/bin/clang -O3 -DNDEBUG  -stdlib=libstdc++ -rdynamic CMakeFiles/MySQL-client-example.dir/main.c.o  -o MySQL-client-example /home/travis/build/ingenue/hunter/_testing/Hunter/_Base/37b04f9/584478b/20e478a/Install/lib/libmysqlclient.a -lpthread -lm -lrt -ldl

vs mine:
    
    /usr/local/clang-3.5.0/bin/clang -O3 -DNDEBUG  -stdlib=libstdc++ -rdynamic CMakeFiles/MySQL-client-example.dir/main.c.o  -o MySQL-client-example -Wl,-rpath,/home/travis/build/caseymcc/hunter/_testing/Hunter/_Base/47252c3/584478b/20e478a/Install/lib /home/travis/build/caseymcc/hunter/_testing/Hunter/_Base/47252c3/584478b/20e478a/Install/lib/libmysqlclient.so

your run has:

    .../libmysqlclient.a -lpthread -lm -lrt -ldl

vs mine

     -Wl,-rpath, .../lib .../libmysqlclient.so

has something changed in the travis/poly setup?

**Comment by caseymcc** [comment]
also not sure why yours compiled a static while mine compiled shared object.

**Comment by ruslo** [comment]
The static build is a right approach. I guess build with dynamic library was just masking some issue.

**Comment by caseymcc** [comment]
Finally got around to looking at this again, I got it fixed.

https://travis-ci.org/caseymcc/hunter/builds/253264215
https://ci.appveyor.com/project/caseymcc/hunter/build/1.0.136

However linking the static version is the issue to start with. Apparently, before it was linking against the shared version by default (as both static and shared are built by the cmake setup unless specifically turned off) see previous version linking below:

    /usr/bin/gcc -O3 -DNDEBUG  -rdynamic CMakeFiles/MySQL-client-example.dir/main.c.o  -o MySQL-client-example -Wl,-rpath,/home/travis/build/caseymcc/hunter/_testing/Hunter/_Base/9e75150/584478b/266cf4a/Install/lib /home/travis/build/caseymcc/hunter/_testing/Hunter/_Base/9e75150/584478b/266cf4a/Install/lib/libmysqlclient.so

my change made it use the static lib which from what I gather can have issues. I found a few references to the issue but I think this link from the mariadb sums up the uniqueness of the issue (under Sergei Golubchik's comment).

https://mariadb.atlassian.net/browse/MDEV-5474?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel

basically boils down to a c library being called from a c++ library and having no access to the std c++ library (but varying greatly on the compiler options used). I don't see anywhere where it has been fixed or even that it is considered a real problem (some people mention adding -lstd++ to the linking can fix it) but most of what I have read about the MySQL lib is that the shared version is considered the one to use.

So if setting the shared version up as the default is ok, I will make the modification needed to finish off this PR.

**Comment by ruslo** [comment]
> So if setting the shared version up as the default is ok, I will make the modification needed to finish off this PR

Okay, please do.

**Comment by caseymcc** [comment]
Ok, should be good now.

https://travis-ci.org/caseymcc/hunter/builds/261099654
https://ci.appveyor.com/project/caseymcc/hunter/build/1.0.140

File: cmake/projects/MySQL-client/hunter.cmake [hunk 1]
```diff
@@ -31,6 +31,17 @@ hunter_add_version(
...
+    VERSION
+    "6.1.9-p0"
+    URL
+    "https://github.com/hunter-packages/mysql-client/archive/v1.6.9-p0.tar.gz"
+    SHA1
+    64c3fb0500e03fc0ee188dca1ec33e52c2f54e32
+)
+
 # https://github.com/ruslo/hunter/issues/705
 # hunter_cacheable(MySQL-client)
```
Comment (ruslo) [inline cmake/projects/MySQL-client/hunter.cmake]: Is it cacheable now?

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2107
* https://travis-ci.org/ingenue/hunter/builds/261254631?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Package not cacheable - no upload stage. Released: https://github.com/ruslo/hunter/releases/tag/v0.19.58
